qemu-cc/qemu-cc.spec

```diff
@@ -0,0 +1,214 @@
+Name: qemu-cc
+Version: v2.9.0+git.a2e6957b6f
+Release: 0
+Source0: %{name}-%{version}.tar.gz
+Source1: qemu-cc-rpmlintrc
+BuildRoot: %{_tmppath}/%{name}-%{version}-%{release}
+
+Summary  : OpenBIOS development utilities
+Group    : Development/Tools
+License  : BSD-2-Clause BSD-3-Clause GPL-2.0 GPL-2.0+ LGPL-2.0+ LGPL-2.1
+
+Requires: qemu-cc-bin
+Requires: qemu-cc-data
+
+BuildRequires : automake
+BuildRequires : bison
+BuildRequires : flex
+BuildRequires : glib2-devel
+BuildRequires : libattr-devel
+BuildRequires : libcap-devel
+BuildRequires : libcap-ng-devel
+BuildRequires : libtool
+BuildRequires : libtool-ltdl-devel
+BuildRequires : libtool
+BuildRequires : m4
+BuildRequires : numactl-devel
+BuildRequires : python-devel
+BuildRequires : zlib
+BuildRequires : pkgconfig(pixman-1)
+Patch1: configure.patch
+
+%undefine _missing_build_ids_terminate_build
+%define  debug_package %{nil}
+
+%description
+===========
+QEMU is a generic and open source machine & userspace emulator and
+virtualizer.
+
+%package bin
+Summary: bin components for the qemu-cc package.
+Group: Binaries
+Requires: qemu-cc-data
+
+%description bin
+bin components for the qemu-cc package.
+
+
+%package data
+Summary: data components for the qemu-cc package.
+Group: Data
+
+%description data
+data components for the qemu-cc package.
+
+%package doc
+Summary: docs for the qemu-cc package.
+Group: Data
+
+%description doc
+Docs for the qemu-cc package.
+
+%prep
+%setup -q
+%patch1 -p1
+
+%build
+export LANG=C
+%configure --prefix=/usr \
+	   --datadir=/usr/share/qemu-cc \
+	   --libdir=/usr/lib64/qemu-cc \
+	   --libexecdir=/usr/libexec/qemu-cc \
+	   --disable-tools \
```

Comment:
We've got to be careful here for various reasons...

Here's how `qemu-lite` is built in clr:

- https://github.com/clearlinux-pkgs/qemu-lite/blob/master/configure

That is almost identical to `qemu-lite.spec` except that that spec file adds two extra options:

```
$ comm -3 clr-qemu-lite.configure qemu-lite.spec 
        --disable-docs
        --disable-static
```

It should be safe to add those two to `qemu-cc.spec`.

Comparing `qemu-lite.spec` and `setup_env_ubuntu.sh`, it looks like `setup_env_ubuntu.sh` is missing:

```
--disable-uuid (I think this is valid as we always specify a value to `-uuid`, so qemu doesn't need to generate one).
--enable-kvm (this is admittedly the default)
--enable-vhost-net (needed!)
--disable-qom-cast-debug
```

Also -- oddly -- `setup_ubuntu_env.sh` adds the following...

```
--disable-cap-ng
--disable-attr
--enable-trace-backend=nop
```
I had thought that those disabled options were required for 9p. @grahamwhaley - do you have any input on that? I'm also not sure why it is explicitly enabling a trace backend, unless that effectively "disables" tracing which is enabled by default (the name suggests it might).

Finally, `setup_ubuntu_env.sh` doesn't specify any `CFLAGS`. That's fine for a CI build, but we need them for packaged builds for security reasons.

So, in summary, I think you need to remove the following from `qemu-cc.spec`:

```
--disable-attr
--disable-cap-ng
```
... add the following to `qemu-cc.spec`:

```
--disable-docs
--disable-qom-cast-debug
--disable-static
--disable-strip
--disable-uuid
--enable-attr
--enable-kvm
--enable-vhost-net
--enable-virtfs
--extra-cflags="-fno-semantic-interposition -O3 -falign-functions=32 -D_FORTIFY_SOURCE=2 -fPIE"
--extra-ldflags="-pie"
```

For the `cflags` options, see:

- https://github.com/01org/cc-oci-runtime/pull/775
- https://github.com/01org/cc-oci-runtime/pull/778

@sboeuf, @mcastelino, @grahamwhaley, @anthonyzxu - please can you review this?